apps/emqx_authz/src/emqx_authz_schema.erl

```diff
@@ -221,6 +223,13 @@ check_headers(Conf) ->
     Headers = hocon_schema:get_value("config.headers", Conf),
     Method =:= <<"post">> orelse (not maps:is_key(<<"content-type">>, Headers)).
 
+check_url(RawURL) ->
+    URL = emqx_authz:http_fullpath_uri_encode(RawURL),
```

Comment:
IMO, there is no need to make things complicated.
make it simple: substitute placeholders with some test values and then validate the replaced string.